pennylane/measurements.py

```diff
@@ -445,16 +591,31 @@ def circuit(x):
             f"{op.name} is not an observable: cannot be used with sample"
         )
 
+    try:
+        eigvals = qml.eigvals(op)
```

Comment:
The cost of using the qml.eigvals every time is surely greater than what it brings. In what cases are eigenvalues returned as integers? Hardcoding in the observables is probably the easiest solution as Josh suggested.

Reply:
That's a fair point. I've updated the logic to consider the built-in observables with integer eigenvalues (https://github.com/PennyLaneAI/pennylane/pull/2044/commits/2393f68e12646d999aeb5e30f86d5bebe2d4c7ac).

> In what cases are eigenvalues returned as integers?

It seems that the observables that depend on `from pennylane.utils import pauli_eigs` and `qml.Identity` have integer eigenvalues. Overall, having integers could be useful for users who would like to utilize reduced memory usage (i.e., storing samples as integers instead of floats). However, if that's not a widely used use case, we could think about making all the observable eigenvalues floats.